numpy/lib/tests/test_histograms.py

```diff
@@ -502,15 +509,16 @@ def test_novariance(self):
 
     def test_limited_variance(self):
         """
-        Check when IQR is 0, but variance exists, we return the sturges value
-        and not the fd value.
+        Check when IQR is 0, but variance exists, we return a reasonable value.
         """
         lim_var_data = np.ones(1000)
         lim_var_data[:3] = 0
         lim_var_data[-4:] = 100
 
         edges_auto = histogram_bin_edges(lim_var_data, 'auto')
-        assert_equal(edges_auto, np.linspace(0, 100, 12))
+        assert_equal(edges_auto[0], 0)
+        assert_equal(edges_auto[-1], 1)
```

Comment:
```suggestion
        assert_equal(edges_auto[-1], 100.)
```
Will just apply it to have passing tests, but feel free to force push it away.